fractal_server/app/runner/v2/__init__.py

```diff
@@ -114,10 +115,33 @@ async def submit_workflow(
 
     with next(DB.get_sync_db()) as db_sync:
 
-        job: JobV2 = db_sync.get(JobV2, job_id)
-        if not job:
+        try:
+            job: Optional[JobV2] = db_sync.get(JobV2, job_id)
+            dataset: Optional[DatasetV2] = db_sync.get(DatasetV2, dataset_id)
+            workflow: Optional[WorkflowV2] = db_sync.get(
+                WorkflowV2, workflow_id
+            )
+        except Exception as e:
+            raise RuntimeError(
```

Comment:
We cannot `raise` from here. This should be a `logger.error`, followed by `reset_logger_handlers(logger)`.